server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -752,7 +768,7 @@ public void deleteStaleSegments(int lastNMetadataFilesToKeep) throws IOException
         );
         List<String> metadataFilesToBeDeleted = metadataFilesEligibleToDelete.stream().filter(metadataFile -> {
             try {
-                return !isLockAcquired(metadataFile);
+                return !isLockAcquired(metadataFile, useCache);
             } catch (IOException e) {
                 logger.error(
```

Comment:
Lets please evaluate if we can get a list of all locks acquired in one call rather than calling list for each metadata file.

Reply:
Let me think more on this. We can list all the entries from `LockDirectory` but need to check if it can be scalable if we start locking metadata files for some other use case.